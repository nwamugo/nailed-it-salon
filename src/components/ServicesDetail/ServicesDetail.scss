.services-detail-container {
  width: 100%;
  height: 30rem;
  display: flex;
  justify-content: space-between;

  @media (max-width: 575.98px) {
    // width: 100%;
    display: flex;
    height: 28.571rem;
    .services-image--manicure,
    .services-image--pedicure,
    .services-image--add-ons,
    .services-nail-extensions {
      display: none;
    }
  }
  @media (min-width: 576px) and (max-width: 767.98px) {
    // width: 50%;
    display: flex;
    height: 28.571rem;
    .services-image--manicure,
    .services-image--pedicure,
    .services-image--add-ons,
    .services-nail-extensions {
      display: none;
    }
  }
  @media (min-width: 768px) and (max-width: 991.98px) {
    // width: 50%;
    display: flex;
    height: 28.571rem;
    .services-image--manicure,
    .services-image--pedicure,
    .services-image--add-ons,
    .services-nail-extensions {
      display: none;
    }
  }
  @media (max-width: 1024px) {
    // width: 50%;
    display: flex;
    height: 28.571rem;
    .services-image--manicure,
    .services-image--pedicure,
    .services-image--add-ons,
    .services-nail-extensions {
      display: none;
    }
  }
}
.services-name {
  font-size: 2.8rem;
}

.services-name-border-top {
  width: 4.72rem;
  height: 0.288rem;
  margin: 0 auto 3rem;

  &--manicure,
  &--add-ons {
    background-color: #000;
  }

  &--pedicure,
  &--nail-extensions {
    background-color: #fff;
  }
}

.services-description {
  flex-grow: 1;
  font-family: "Libre Baskerville";
  text-align: center;
  padding-top: 6rem;

  &--manicure,
  &--add-ons {
    background-color: #f8e7f2;
    color: #000;
  }

  &--pedicure,
  &--nail-extensions {
    background-color: #000;
    color: #fff;
  }

  &--nail-extensions {
    order: 1;
  }
}

.ticketImage {
  width: 35.4rem;
  height: 100%;
  object-fit: cover;

  @media (max-width: 575.98px) {
    display: None;
  }
  @media (min-width: 576px) and (max-width: 767.98px) {
    display: None;
  }
  @media (min-width: 768px) and (max-width: 991.98px) {
    display: None;
  }
  @media (max-width: 1024px) {
    display: None;
  }
}

.btn-book-now {
  border: none;
  border-radius: 1rem;
  cursor: pointer;
  margin-top: 6rem;
  padding: 0.8rem 1.6rem 0.8rem 1.6rem;
  text-transform: uppercase;

  &--manicure,
  &--add-ons {
    background-color: #e294c5;
    color: #fff;
  }

  &--manicure:hover,
  &--add-ons:hover {
    background-color: #000;
    color: #e294c5;
  }

  &--pedicure,
  &--nail-extensions {
    background-color: #000;
    color: #e294c5;
  }

  &--pedicure:hover,
  &--nail-extensions:hover {
    background-color: #e294c5;
    color: #fff;
  }
}
.button-text {
  font-size: 1.8rem !important;
}
