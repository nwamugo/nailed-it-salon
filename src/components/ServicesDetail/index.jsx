import React from 'react';

import './ServicesDetail.scss';

const ServicesDetail = ({ image, cardTitle }) => {
  const appendTitle = cardTitle.toLowerCase().replace(' ', '-');

  return (
    <div className="services-detail-container">
      <div className={`hidden-xs services-image--${appendTitle}`}>
        <img
          data-test="logoIMG"
          src={image}
          alt="Logo"
          className="ticketImage"
        />
      </div>
      <div
        className={`services-description services-description--${appendTitle}`}
      >
        <hr
          className={`services-name-border-top services-name-border-top--${appendTitle}`}
        />
        <h2 className={`services-name services-name--${appendTitle}`}>
          {cardTitle}
        </h2>

        <button
          className={`btn-lg btn-book-now btn-book-now--${appendTitle}`}
          type="submit"
        >
          <span className="button-text">Book Now &gt;</span>
        </button>
      </div>
    </div>
  );
};

export default ServicesDetail;
