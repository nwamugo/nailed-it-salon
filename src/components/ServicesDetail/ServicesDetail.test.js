import React from 'react';
import { shallow } from 'enzyme';
import ServicesDetail from './index';

describe('ServicesDetail', () => {
  it('should not have regressed', () => {
    const component = shallow(<ServicesDetail cardTitle="TestingComponent" />);

    expect(component).toMatchSnapshot();
  });
});
