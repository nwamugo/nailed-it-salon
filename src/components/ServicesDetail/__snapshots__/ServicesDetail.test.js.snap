// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesDetail should not have regressed 1`] = `
<div
  className="services-detail-container"
>
  <div
    className="hidden-xs services-image--testingcomponent"
  >
    <img
      alt="Logo"
      className="ticketImage"
      data-test="logoIMG"
    />
  </div>
  <div
    className="services-description services-description--testingcomponent"
  >
    <hr
      className="services-name-border-top services-name-border-top--testingcomponent"
    />
    <h2
      className="services-name services-name--testingcomponent"
    >
      TestingComponent
    </h2>
    <button
      className="btn-lg btn-book-now btn-book-now--testingcomponent"
      type="submit"
    >
      <span
        className="button-text"
      >
        Book Now &gt;
      </span>
    </button>
  </div>
</div>
`;
